book/src/smart-contracts-architecture/utils-module.md

```diff
@@ -4,24 +4,42 @@ The Utils module is a various collection of utility functions and tools designed
 
-- [u128_mask.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/keys.cairo): Mask function.
+- [global_reentrancy_guard.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/global_reentrancy_guard.cairo): This file implements a global-level reentrancy guard to secure smart contract functions against reentrancy attacks. It uses a global flag, `REENTRANCY_GUARD_STATUS`, to indicate whether a protected function is currently executing, and provides `non_reentrant_before` and `non_reentrant_after` methods to set and reset this flag, preventing malicious reentrant calls and ensuring consistent state throughout the smart contract's execution.
 
-- [validate_account.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/keys.cairo): Helps validating accounts.
+- [hash.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/hash.cairo): This file provides utility functions for hashing, specifically using the Poseidon hash function. It includes a function, `hash_poseidon_single`, which allows hashing a single `felt252` value using Poseidon, ensuring data integrity and catering to cryptographic needs in the smart contract.
+
+- [i128_test_storage_contract.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/i128_test_storage_contract.cairo): This file defines a StarkNet contract intended to test the storage and retrieval of `i128` values in the contract state. It serves as a test utility to ensure correct handling of `i128` values in the smart contract environment.
+
+- [i128.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/i128.cairo): This file provides implementations for handling `i128` (signed 128-bit integers) in Cairo, including arithmetic operations, serialization, deserialization, and storage management, enabling the efficient and accurate handling of `i128` values in StarkNet smart contracts.
+
+- [precision.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/precision.cairo): Provides utility functions for high precision arithmetic and conversion operations, facilitating precise calculations and conversions between different units, such as from float to wei, applying factors, and handling rounding in the Satoru StarkNet smart contract environment.
+
+- [span32.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/span32.cairo): Provides utility functions for managing and manipulating fixed-size arrays (span32) and facilitating serialization and deserialization of these arrays, including handling of storage reads and writes in the Satoru StarkNet smart contract environment. 
```

Comment:
```suggestion
- [span32.cairo](https://github.com/keep-starknet-strange/satoru/blob/main/src/utils/span32.cairo): Provides utility functions for managing and manipulating fixed-size arrays (span32). A wrapper around Span type with a maximum size of 32. Used to prevent size overflow when storing Span. 
```